userspace/libsinsp/test/filterchecks/evt.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright (C) 2023 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <test/helpers/threads_helpers.h>
+
+TEST_F(sinsp_with_test_input, EVT_FILTER_is_open_create)
+{
+	add_default_init_thread();
+
+	open_inspector();
+
+	std::string path = "/home/file.txt";
+	int fd = 3;
+
+	// In the enter event we don't send the `PPM_O_F_CREATED`
+	sinsp_evt* evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_OPEN_E, 3, path.c_str(),
+					      (uint32_t)PPM_O_RDWR | PPM_O_CREAT, (uint32_t)0);
+	ASSERT_EQ(get_field_as_string(evt, "evt.is_open_create"), "false");
+
+	// The `fdinfo` is not populated in the enter event
+	ASSERT_FALSE(evt->m_fdinfo);
+
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_OPEN_X, 6, fd, path.c_str(),
+				   (uint32_t)PPM_O_RDWR | PPM_O_CREAT | PPM_O_F_CREATED, (uint32_t)0, (uint32_t)5,
+				   (uint64_t)123);
+	ASSERT_EQ(get_field_as_string(evt, "evt.is_open_create"), "true");
```

Comment:
So, emscripten tests are failing on this line: 
```
[ RUN      ] sinsp_with_test_input.EVT_FILTER_is_open_create
/home/runner/work/libs/libs/userspace/libsinsp/test/filterchecks/evt.cpp:40: Failure
Expected equality of these values:
  get_field_as_string(evt, "evt.is_open_create")
    Which is: "false"
  "true"
```
So, it seems like the filtercheck:
```
if(m_field_id == TYPE_ISOPEN_CREATE &&
				   flags & PPM_O_F_CREATED)
				{
					m_u32val = 1;
				}
```
is not seeing the `PPM_O_F_CREATED` bit in `flags`.

My guess is that somehow parsers code is replacing our flags with the enter event ones (since you used the same `tid == 1` in both `add_event_advance_ts` calls), but it is not seeing the current flags having the `PPM_O_F_CREATED` bit enabled:
```
// keep PPM_O_F_CREATED flag if present
				if (flags & PPM_O_F_CREATED)
					flags = enter_evt_flags | PPM_O_F_CREATED;
				else
					flags = enter_evt_flags; // i think we fall in this branch
```

That is really weird, though. And why does it happen only on emscripten builds?

Reply:
I've tried doing some debugging and found something weird...
```c
	if(etype == PPME_SYSCALL_OPEN_X)
	{
		parinfo = evt->get_param(1);
		name = parinfo->m_val;
		namelen = parinfo->m_len;

		parinfo = evt->get_param(2);
		ASSERT(parinfo->m_len == sizeof(uint32_t));
		flags = *(uint32_t *)parinfo->m_val;

		printf("Exit event:\n");
		printf("	name: %s\n", name);
		printf("	namelen: %d\n", namelen);
		printf("	flags: %d\n", flags);

                ...

		if(lastevent_retrieved && enter_evt->get_num_params() >= 2)
		{
			parinfo = enter_evt->get_param(0);
			enter_evt_name = parinfo->m_val;
			enter_evt_namelen = parinfo->m_len;

			parinfo = enter_evt->get_param(1);
			ASSERT(parinfo->m_len == sizeof(uint32_t));
			enter_evt_flags = *(uint32_t *)parinfo->m_val;

			printf("Enter event:\n");
			printf("	name: %s\n", enter_evt_name);
			printf("	namelen: %d\n", enter_evt_namelen);
			printf("	flags: %d\n", enter_evt_flags);

                        ...

```

On `amd64` exit event values are the same as enter event, except for `flags` that will be updated after with the new check.

```
[ RUN      ] sinsp_with_test_input.EVT_FILTER_is_open_create
Exit event:
	name: /home/file.txt
	namelen: 15
	flags: 16391
Enter event:
	name: /home/file.txt
	namelen: 15
	flags: 7
```

But on `emscripten` `name` and `flag` values are wrong on the exit event:

```
[ RUN      ] sinsp_with_test_input.EVT_FILTER_is_open_create
Exit event:
	name: _subreaper
	namelen: 11
	flags: 0
Enter event:
	name: /home/file.txt
	namelen: 15
	flags: 7
```

I can't figure out where that name comes from, and as you can see exit flags is `0`, so obviously the test fails.
The other tests are ok.

Reply:
This should solve the issue: https://github.com/falcosecurity/libs/pull/1299#discussion_r1304082744

Reply:
Nevermind, @Andreagit97 found the fix! 🎉